sentinelhub/download/handlers.py

```diff
@@ -58,6 +58,9 @@ def new_download_func(self, request):
                 time.sleep(sleep_time)
                 sleep_time *= backoff_coefficient
 
+        raise DownloadFailedException('No download attempts available - configuration parameter max_download_attempts '
```

Comment:
Any chance this part is reached because of some other (unhandled) exception? E.g. the `download_func` fails with "disk full" or sth? In such case the message is misleading.

Reply:
Hm, good question but I'm quite sure that it is not possible. Our try-except is only catching `requests.RequestException` errors. Anything else would not be caught and therefore raise an error. The `except` segment also makes sure that it raises an error at the end of the last attempt. So the only way that the code can go past the `for` loop is if it never enters it.